staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -671,6 +673,29 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+// Returns true if the given err is "connection refused" error.
+func IsConnectionError(err error) bool {
+	if net.IsConnectionReset(err) {
+		return true
+	}
+	if urlErr, ok := err.(*url.Error); ok {
+		err = urlErr.Err
+	}
+	if osErr, ok := err.(*os.SyscallError); ok {
+		err = osErr.Err
+	}
+	if errno, ok := err.(syscall.Errno); ok && errno == syscall.ECONNREFUSED {
+		return true
+	}
+	return false
+}
+
+// isAPIServerShutdown checks if the given error contains "apiserver is shutting down" string.
+// TODO: This error should probably get a typed error.
+func isAPIServerShutdown(err error) bool {
+	return strings.Contains(err.Error(), "apiserver is shutting down")
```

Comment:
we control the server here... rather than making this check stringly typed, if we want this to retry, I'd expect us to write an error with a retry-after value here:

https://github.com/kubernetes/kubernetes/blob/b61baea13594f8cacd8e7db86aa68cbd967b9a5e/staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go#L41

Reply:
fixed.